uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1018.php

```diff
@@ -27,11 +24,30 @@ public function upgrade_1018 () {
     $result = CRM_Core_DAO::executeQuery($query)->fetchAll();
     $absenceTypesInHour = array_column($result, 'id');
 
+    if (count($absenceTypesInHour) > 0) {
+      $this->update_single_day_requests_in_hours_end_time(
+        implode(',', $absenceTypesInHour));
+    }
+
+    return true;
+  }
+
+  /**
+   * Fetches single day leave requests in hours,
+   * retrieves information about work pattern day based on
+   * the date of the leave request and the contact and,
+   * if it is still a working day, sets the end time from
+   * the working day as an end time of the leave request.
+   *
+   * @param array $absenceTypesInHourIDs - comma separated list of IDs of
+   *   Absence Types with "hours" calculation unit
+   */
+  private function update_single_day_requests_in_hours_end_time ($absenceTypesInHourIDs) {
```

Comment:
For private methods in Upgraders we use this pattern: `up<upgrade number>_nameInCamelCase()` (check [here](https://github.com/civicrm/civihr/blob/528028facc466128e44cf84116926b9d14cadfc8/uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1001.php#L71) for an example). The reason is to avoid collisions with other traits or private methods in the classes using the trait.

If you don't mind having two returns in this method another possible implementation would be:

```php
if (empty($absenceTypesInHours)) {
   return true;
}

... do the update

return true;
```